python/ray/serve/_private/replica.py

```diff
@@ -289,18 +349,74 @@ def get_num_ongoing_requests(self) -> int:
         This runs on a separate thread (using a Ray concurrency group) so it will
         not be blocked by user code.
         """
-        return self._queue_metrics_manager.get_num_pending_and_running_requests()
+        return self._metrics_manager.get_num_pending_and_running_requests()
+
+    @contextmanager
+    def _wrap_user_method_call(self, request_metadata: RequestMetadata):
```

Comment:
Kinda curious are there some changes to this so that it doesn't need to be async anymore?

Reply:
actually it didn't need to be async before... there were no `await` calls to begin with :)